docs/docsite/rst/contributor/DJANGO_REQUIREMENTS.rst

```diff
@@ -0,0 +1,843 @@
+=====================================================
+Django Development Requirements
+=====================================================
+
+**AWX Codebase Best Practices**
+
+:Version: 1.0
+:Date: September 2025
+:Based on: AWX Enterprise Django Application Analysis
+:Generated by: Claude Code AI 
+
+----
+
+.. contents:: Table of Contents
+   :depth: 3
+   :local:
+
+----
+
+1. Project Structure
+====================
+
+1.1 Modular Application Architecture
+------------------------------------
+
+**REQUIRED**: Organize Django project with clear separation of concerns::
+
+    awx/
+    ├── __init__.py              # Version management and environment detection
+    ├── main/                    # Core business logic and models
+    ├── api/                     # REST API layer (Django REST Framework)
+    ├── ui/                      # Frontend integration
+    ├── conf/                    # Configuration management
+    ├── settings/                # Environment-specific settings
+    ├── templates/               # Django templates
+    └── static/                  # Static assets
+
+**Requirements**:
+
+- Each functional area must have its own Django app
+- Use descriptive app names that reflect business domains
+- Implement proper ``__init__.py`` files for package initialization
+- Separate API logic from core business logic
+
+1.2 Custom Management Commands
```

Comment:
This section is less about "Custom Management Commands" and more about "Code that Runs _Before_ Every Management Command"